src/microbe_stage/systems/MicrobeEmissionSystem.cs

```diff
@@ -209,6 +216,48 @@ private static Vector3 FacingDirection(in Entity entity, ref WorldPosition posit
         ++control.FiredToxinCount;
     }
 
+    private void EmitIron(in Entity entity, ref MicrobeControl control, ref OrganelleContainer organelles,
+        ref CellProperties cellProperties, ref SoundEffectPlayer soundEffectPlayer, ref WorldPosition position,
+        bool engulfed)
+    {
+        if (engulfed)
+            return;
+
+        if (control.AgentEmissionCooldown > 0)
+            return;
+
+        // Only shoot if you have any iron organelles
+        if (organelles.IronBreakdownEfficiency < 1)
+            return;
+
+        // Can't shoot if membrane is not ready
+        if (!cellProperties.IsMembraneReady())
+            return;
+
+        // The cooldown time is inversely proportional to the amount of agent vacuoles with a little nerf.
+        control.AgentEmissionCooldown = Constants.AGENT_EMISSION_COOLDOWN * 5 / organelles.IronBreakdownEfficiency;
+
+        float ejectionDistance = cellProperties.CreatedMembrane!.EncompassingCircleRadius +
+            Constants.AGENT_EMISSION_DISTANCE_OFFSET;
+
+        if (cellProperties.IsBacteria)
+            ejectionDistance *= 0.5f;
+
+        // Find the direction the microbe is facing
+        // (actual rotation, not LookAtPoint, also takes colony membership into account and uses the
+        // parent rotation)
+        var direction = FacingDirection(entity, ref position);
```

Comment:
This position etc. logic is duplicated from EmitToxin so this should be split into its own method so that there is less code copy-paste.

Reply:
I don't think that's necessary (for me)

Reply:
? If we still had the duplicate code check it would complain about this. This has quite a lot of complex math that is duplicated. If someone writes a fix for example for cell colonies no longer hitting their own toxins, there's a chance the bug is fixed only in one place, which is why code duplication is a serious problem in terms of codebase maintainability.